### PR TITLE
add net.gwyddion.Gwyddion

### PR DESCRIPTION
**Pull request by Hofer-Julian**

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

File: net.gwyddion.Gwyddion.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home"
+    ],
+    "modules": [     
+        "shared-modules/gtk2/gtk2.json",  
+        {
+            "name": "gwyddion",
+            "buildsystem": "autotools",
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]:
```suggestion
```
Already the default one

File: net.gwyddion.Gwyddion.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	    "config-opts": [
+		
+	    ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://sourceforge.net/projects/gwyddion/files/gwyddion/2.53/gwyddion-2.53.tar.xz",
+                    "md5": "310733ff22ed978912195d5ac60b3bae"
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]: Any reasons for using a md5 hash?

File: net.gwyddion.Gwyddion.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=home"
+    ],
+    "modules": [     
+        "shared-modules/gtk2/gtk2.json",  
+        {
+            "name": "gwyddion",
+            "buildsystem": "autotools",
+	    "config-opts": [
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]: Empty config, you can remove it.

File: net.gwyddion.Gwyddion.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+    "app-id": "net.gwyddion.Gwyddion",
+    "branch": "stable",
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]:
```suggestion
```
Flathub already sets that for you.

File: net.gwyddion.Gwyddion.json (hunk 1 above)
Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Then I'll remove it :)

File: net.gwyddion.Gwyddion.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.32",
+    "sdk": "org.gnome.Sdk",
+    "command": "gwyddion",
+    "rename-icon": "gwyddion",
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]: I would use a copy-icon: true too. Maybe the icon is used somewhere else with the old name.

File: net.gwyddion.Gwyddion.json (hunk 2 above)
Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: What alternative do I have?
https://sourceforge.net/projects/gwyddion/files/gwyddion/2.53/

File: net.gwyddion.Gwyddion.json (hunk 3 above)
Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Okay

**Comment by bilelmoussaoui** [comment]
Build fails for a missing appdata file. 
```
stat: cannot stat 'builddir/*/share/app-info/xmls/net.gwyddion.Gwyddion.xml.gz': No such file or directory
```

File: net.gwyddion.Gwyddion.json (hunk 5 above)
Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Okay!

**Comment by Hofer-Julian** [comment]
> Build fails for a missing appdata file.
> 
> ```
> stat: cannot stat 'builddir/*/share/app-info/xmls/net.gwyddion.Gwyddion.xml.gz': No such file or directory
> ```

Yeah, still trying to find out, where the appdata is

File: net.gwyddion.Gwyddion.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+    "app-id": "net.gwyddion.Gwyddion",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.32",
+    "sdk": "org.gnome.Sdk",
+    "command": "gwyddion",
+    "copy-icon": "true",
```
Comment (bilelmoussaoui) [inline net.gwyddion.Gwyddion.json]:
```suggestion
    "copy-icon": true,
```

**Comment by bilelmoussaoui** [comment]
> > Build fails for a missing appdata file.
> > 
> >   
> >     
> >       
> >     
> >   ```
> > stat: cannot stat 'builddir/*/share/app-info/xmls/net.gwyddion.Gwyddion.xml.gz': No such file or directory
> > ```
> 
> Yeah, still trying to find out, where the appdata is

They don't seem to ship one :/

File: net.gwyddion.Gwyddion.json (hunk 2 above)
Comment (nedrichards) [reply net.gwyddion.Gwyddion.json]: From project described hashes, none! You could manually run sha256sum or sha512sum on the file though if you wanted.

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by Hofer-Julian** [comment]
It seems like there are still problems with strict validation:
```
• attribute-invalid     : <screenshot> width too small [http://gwyddion.net/screenshots/gwyddion-screenshot-0s.jpeg] minimum is 624px
• attribute-invalid     : <screenshot> height too small [http://gwyddion.net/screenshots/gwyddion-screenshot-0s.jpeg] minimum is 351px
• attribute-invalid     : <screenshot> width too large [http://gwyddion.net/screenshots/gwyddion-screenshot-3.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [http://gwyddion.net/screenshots/gwyddion-screenshot-3.png] maximum is 900px
• attribute-invalid     : <screenshot> width too large [http://gwyddion.net/screenshots/gwyddion-screenshot-2.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [http://gwyddion.net/screenshots/gwyddion-screenshot-2.png] maximum is 900px
• attribute-invalid     : <screenshot> width too small [http://gwyddion.net/screenshots/gwyddion-screenshot-1s.jpeg] minimum is 624px
• attribute-invalid     : <screenshot> height too small [http://gwyddion.net/screenshots/gwyddion-screenshot-1s.jpeg] minimum is 351px
```
Does that mean there the screenshots will not be displayed?

**Comment by nedrichards** [comment]
No, the flathub.org website has less strict requirements than the strict validation. If the build passes then it should show up fine.

**Comment by Hofer-Julian** [comment]
@nedrichards 👍

**Comment by nedrichards** [comment]
@Hofer-Julian are you part of the upstream project? If not then you need to let them know that you'd like to add it and ask if they'd like to be involved and have commit access. It won't block things if they don't want to be, but flathub is designed to be a place for developers to reach users directly.

**Comment by nedrichards** [comment]
@Hofer-Julian also, you should try and send the appdata upstream as this adds value to every distribution on Linux.

**Comment by Hofer-Julian** [comment]
@nedrichards I am not part of the upstream project. Please see this https://sourceforge.net/p/gwyddion/discussion/general/thread/58bd4e52a4/?limit=25#697a

File: net.gwyddion.Gwyddion.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    "sdk": "org.gnome.Sdk",
+    "command": "gwyddion",
+    "copy-icon": true,
+    "rename-icon": "gwyddion",
+    "rename-desktop-file": "gwyddion.desktop",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (nedrichards) [inline net.gwyddion.Gwyddion.json]: does this *actually* work under wayland? It's GTK2 right?

**Comment by Hofer-Julian** [comment]
@nedrichards Good point, will do that

File: net.gwyddion.Gwyddion.json (hunk 7 above)
Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Ah, I forgot to test it under wayland. Will do it right away.

Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: It works. What is it with GTK2 that you assumed otherwise?

Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Ah now I see it. Even under wayland, if use
` flatpak run --nosocket=x11 net.gwyddion.Gwyddion `
it does not run. But if I use 
`flatpak run --nosocket=wayland net.gwyddion.Gwyddion `
everything works.

Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Should I remove "--socket=wayland"?

Comment (nedrichards) [reply net.gwyddion.Gwyddion.json]: You should. It'll run under XWayland.

Comment (Hofer-Julian) [reply net.gwyddion.Gwyddion.json]: Done

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by Hofer-Julian** [comment]
Here is a [statement](https://sourceforge.net/p/gwyddion/discussion/general/thread/58bd4e52a4/#6872) of one of the main authors that he approves of flatpaking it, but has no interest in flathub access. From my side Gwyddion is ready to be published.

**Comment by Hofer-Julian** [comment]
Sorry for the last minute change.
Saw that an (optional) library was actually quite crucial.

**Comment by Hofer-Julian** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by nedrichards** [comment]
@Hofer-Julian you will need to reconfigure that library with the more recent tools that are available in the SDK which know that aarch64 exists. Example: https://github.com/flathub/com.tux4kids.tuxtype/blob/a98c97c04a7b59e11ab324aa4e6665f69008cebb/com.tux4kids.tuxtype.json#L42

**Comment by Hofer-Julian** [comment]
> @Hofer-Julian you will need to reconfigure that library with the more recent tools that are available in the SDK which know that aarch64 exists. Example: https://github.com/flathub/com.tux4kids.tuxtype/blob/a98c97c04a7b59e11ab324aa4e6665f69008cebb/com.tux4kids.tuxtype.json#L42

I don't really know what to do with that...

Also I'm quite surprised that there are problems since I got the gtkglext and its dependencies from https://github.com/flathub/com.anydesk.Anydesk/blob/master/com.anydesk.Anydesk.json.

**Comment by nedrichards** [comment]
anydesk is x86 only https://github.com/flathub/com.anydesk.Anydesk/blob/master/flathub.json I'll patch in what I mean and try a build

**Comment by Hofer-Julian** [comment]
> 
> 
> anydesk is x86 only https://github.com/flathub/com.anydesk.Anydesk/blob/master/flathub.json I'll patch in what I mean and try a build

Thanks

**Comment by nedrichards** [comment]
bot, build net.gwyddion.Gwyddion

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/net.gwyddion.Gwyddion

**Pull request closed without merging**